### ISSUE TITLE
False positive EmptyArrayAccess in loop which adds elements to an array

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#270]
Expected: no errors
Observed:
```
ERROR: EmptyArrayAccess - src/emptyarray.php:11:45 - Cannot access value on empty array variable $merged               
                $merged[$key] = array_merge($merged[$key], $value);                                                    


ERROR: InvalidArgument - src/emptyarray.php:11:45 - Argument 1 of array_merge expects array<mixed, mixed>, empty provided                                                                                                                      
                $merged[$key] = array_merge($merged[$key], $value);                                                    
```

Snippet to reproduce:

```php
<?php
/**
 * @param array[][] $args
 * @return array[]
 */
function get_merged_dict($args) {
    $merged = array();  // This is incorrectly inferred as being guaranteed to be empty, but it is added to within the loop, and element presence is checked
    foreach ($args as $group) {
        foreach ($group as $key => $value) {
            if (isset($merged[$key])) {
                $merged[$key] = array_merge($merged[$key], $value);
            } else {
                $merged[$key] = $value;
            }
        }
    }   
    return $merged;
}
```

**Comment by muglug** [comment]
Thanks – should be fixed in latest, let me know if you see it in your particular usage.
